### PR TITLE
DEVTOOLS-66: Delete directories on push

### PR DESCRIPTION
**Pull request by juancoen**

### CODE REVIEW

File: lib/controllers/pushController.js [hunk 1]
```diff
@@ -57,28 +55,28 @@ module.exports = function (apiPlatformService, BPromise,
...
         return _.filter(remoteFiles, 'isDirectory');
       }
     })
     .then(function () {
-      var result = _.pick(status, 'addedDirectories');
+      result = _.pick(status, 'addedDirectories');
+
+      var deletedDirectories = status.deletedDirectories;
 
+      return BPromise.each(deletedDirectories, function (deletedDirectory) {
```
Comment (fschargorosdky) [inline lib/controllers/pushController.js]: this has to be done after deleting the files in the remote folders.

File: tests/unit/pushController.test.js [hunk 2]
```diff
@@ -149,7 +156,8 @@ describe('pushController', function () {
             currentWorkspace.apiVersion.id
           ).should.be.true();
 
-          apiPlatformServiceStub.createAPIDirectory.calledThrice;
+          apiPlatformServiceStub.createAPIDirectory.calledThrice
+            .should.be.true();
           apiPlatformServiceStub.createAPIDirectory
             .firstCall.calledWithExactly(firstDirectory);
```
Comment (fschargorosdky) [inline tests/unit/pushController.test.js]: could you add the missing should.be.true() to this calledWithExactly?

File: lib/controllers/pushController.js (hunk 1 above)
Comment (juancoen) [inline lib/controllers/pushController.js]: done

File: tests/unit/pushController.test.js (hunk 2 above)
Comment (juancoen) [inline tests/unit/pushController.test.js]: done

File: lib/controllers/pushController.js [hunk 3]
```diff
@@ -90,6 +77,11 @@ module.exports = function (apiPlatformService, BPromise,
         })
         .then(function () {
           result.deleted = status.deleted;
+          return pushDeletedDirectories(workspace, status.deletedDirectories,
```
Comment (fschargorosdky) [inline lib/controllers/pushController.js]: before deleting, sort the directories in descending order:
```
test1/test2/test3
test1/test2
test1
```

Comment (juancoen) [inline lib/controllers/pushController.js]: done

File: tests/unit/pushController.test.js [hunk 4]
```diff
@@ -133,12 +146,16 @@ describe('pushController', function () {
...
         .onThirdCall().returns(BPromise.resolve(thirdDirectoryResult));
 
+      apiPlatformServiceStub.deleteAPIDirectory.onFirstCall()
+        .returns(BPromise.resolve(secondDirToDelete));
+      apiPlatformServiceStub.deleteAPIDirectory.onSecondCall()
+        .returns(BPromise.resolve(firstDirToDelete));
+
       pushController.push()
-        .then(function (output) {
+        .then(function () {
```
Comment (fschargorosdky) [inline tests/unit/pushController.test.js]: you have to assert that the function result is the right one

File: tests/unit/pushController.test.js [hunk 5]
```diff
@@ -269,17 +309,17 @@ describe('pushController', function () {
...
         deleted: []
       };
       localServiceStub.getStatus.returns(BPromise.resolve(status));
-      localServiceStub.getDirectoriesPath
-        .returns(BPromise.resolve([]));
       apiPlatformServiceStub.createAPIDirectory = sinon.stub();
 
       pushController.push()
-        .then(function (output) {
+        .then(function () {
```
Comment (fschargorosdky) [inline tests/unit/pushController.test.js]: you have to assert that the function result is the right one

Comment (juancoen) [inline tests/unit/pushController.test.js]: done

File: tests/unit/pushController.test.js (hunk 4 above)
Comment (juancoen) [inline tests/unit/pushController.test.js]: done

**Comment by fschargorosdky** [comment]
Looks good to me

**Pull request merged**